testng-core/src/main/java/org/testng/JarFileUtils.java

```diff
@@ -112,7 +111,11 @@ private void delete(File f) throws IOException {
     if (f.isDirectory()) {
       for (File c : Objects.requireNonNull(f.listFiles())) delete(c);
     }
-    if (!f.delete()) throw new FileNotFoundException("Failed to delete file: " + f);
+
+    /** * Provide better logging for files temp that fail to be cleaned up. */
+    if (!Files.deleteIfExists(f.toPath())) {
```

Comment:
I think there is no need of comments here because deleteIfExists is definitively better.

Reply:
Removed the comment